mbf_abstract_nav/src/abstract_controller_execution.cpp

```diff
@@ -185,6 +186,43 @@ void AbstractControllerExecution::setVelocityCmd(const geometry_msgs::TwistStamp
   // TODO so there should be no loss of information in the feedback stream
 }
 
+bool AbstractControllerExecution::checkVelocityIgnore(const geometry_msgs::Twist& cmd_velocity)
+{
+  bool ignored_exceed_tolerance = false;
+
+  bool robot_stopped = robot_info_.isRobotStopped(1e-3, 1e-3);
+
+  // compute linear and angular velocity magnitude
+  const double cmd_linear = std::hypot(cmd_velocity.linear.x, cmd_velocity.linear.y);
+  const double cmd_angular = std::abs(cmd_velocity.angular.z);
+
+  bool cmd_is_not_zero = (cmd_linear > 0.01 || cmd_angular > 0.01);
+
+  // no need to check
+  if (!robot_stopped || !cmd_is_not_zero)
+  {
+    first_ignored_time_ = ros::Time::now();
+    return false;
+  }
+
+  // check if robot ignores the cmd_vel
+  double ignored_duration = (ros::Time::now() - first_ignored_time_).toSec();
+  ROS_WARN_THROTTLE(1,
+                    "Robot ignores velocity command for %.2f seconds.\n The cmd_vel being ignored is: (x=%.2f, "
+                    "y=%.2f, w=%.2f)",
+                    ignored_duration, cmd_velocity.linear.x, cmd_velocity.linear.y, cmd_velocity.angular.z);
+
+  if (ignored_duration > robot_ignore_check_tolerance_)
+  {
+    // the robot is ignoring the velocity command more the threshold time
+    ROS_ERROR("Robot is disabled and the time it ignored velocity command exceeded the tolerance time: %.2f seconds",
```

Comment:
F: we don't know why is ignoring vel commands, so better not to make statements like "robot is disabled"  can be off, can be blocked against a wall,,,,  we don't know